narwhals/_spark_like/dataframe.py

```diff
@@ -173,16 +176,87 @@ def sort(
 
         flat_by = flatten([*flatten([by]), *more_by])
         if isinstance(descending, bool):
-            descending = [descending]
+            descending = [descending] * len(flat_by)
```

Comment:
This took me some mental sanity to figure out. I was like "does pyspark do not know how sorting works?" for a good 5 minutes